Sources/PubNub/KMP/KMPPubNub+Listeners.swift

```diff
@@ -35,58 +35,37 @@ extension KMPPubNub {
+      
       switch newStatus {
       case .connected:
-        listener.onStatusChange?(
-          KMPConnectionStatus(
-            category: .connected,
-            error: nil,
-            currentTimetoken: NSNumber(value: pubnub.previousTimetoken ?? 0),
-            affectedChannels: Set(pubnub.subscribedChannels),
-            affectedChannelGroups: Set(pubnub.subscribedChannelGroups)
-          )
-        )
+        category = .connected
+        errorIfAny = nil
       case .disconnected:
-        listener.onStatusChange?(
-          KMPConnectionStatus(
-            category: .disconnected,
-            error: nil,
-            currentTimetoken: NSNumber(value: pubnub.previousTimetoken ?? 0),
-            affectedChannels: Set(pubnub.subscribedChannels),
-            affectedChannelGroups: Set(pubnub.subscribedChannelGroups)
-          )
-        )
-      case .disconnectedUnexpectedly(let error):
-        listener.onStatusChange?(
-          KMPConnectionStatus(
-            category: error.reason == .malformedResponseBody ? .malformedResponseCategory : .disconnectedUnexpectedly,
-            error: error,
-            currentTimetoken: NSNumber(value: pubnub.previousTimetoken ?? 0),
-            affectedChannels: Set(pubnub.subscribedChannels),
-            affectedChannelGroups: Set(pubnub.subscribedChannelGroups)
-          )
-        )
-      case .connectionError(let error):
-        listener.onStatusChange?(
-          KMPConnectionStatus(
-            category: error.reason == .malformedResponseBody ? .malformedResponseCategory : .connectionError,
-            error: error,
-            currentTimetoken: NSNumber(value: pubnub.previousTimetoken ?? 0),
-            affectedChannels: Set(pubnub.subscribedChannels),
-            affectedChannelGroups: Set(pubnub.subscribedChannelGroups)
-          )
-        )
-      case let .subscriptionChanged(channels, groups):
-        listener.onStatusChange?(
-          KMPConnectionStatus(
-            category: .subscriptionChanged,
-            error: nil,
-            currentTimetoken: NSNumber(value: pubnub.previousTimetoken ?? 0),
-            affectedChannels: Set(channels),
-            affectedChannelGroups: Set(groups)
-          )
-        )
+        category = .disconnected
+        errorIfAny = nil
+      case let .disconnectedUnexpectedly(error):
+        category = .disconnectedUnexpectedly
+        errorIfAny = error
+      case .subscriptionChanged:
+        category = .subscriptionChanged
+        errorIfAny = nil
+      case let .connectionError(error):
+        category = .connectionError
+        errorIfAny = error
       }
+      
+      listener.onStatusChange?(
+        KMPConnectionStatus(
+          category: category,
+          error: errorIfAny,
+          currentTimetoken: NSNumber(value: pubnub.previousTimetoken ?? 0),
+          affectedChannels: Set(pubnub.subscribedChannels),
+          affectedChannelGroups: Set(pubnub.subscribedChannelGroups)
```

Comment:
Do we need to use `channels` and `groups` from `.subscriptionChanged(channels, groups)` as it was in previous commit? If it is important, then maybe two more vars which initially will hold `pubnub.subscribedChannels` and `pubnub.subscribedChannelGroups` and only for `.subscriptionChanged` to be updated??

Reply:
Yup, I'll propose a fix for that and let you know

Reply:
It's not feasible as far as I can see. Kotlin uses [PNStatus](https://github.com/pubnub/kotlin/blob/master/pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/PNStatus.kt#L10-L11) as a container not only for the `PNSubscriptionChanged` category. Therefore, I had to fill those two fields (`affectedChannels`, `affectedChannelGroups`) to adjust to their contract

Reply:
I'm not saying that we shouldn't. I was asking about values which you assign. Previous change in this file looked logical for me because for `.subscriptionChanged(channels, groups)` event, value of `affectedChannels` and `affectedChannelGroups` has been set to `channels` and `groups` from event. Now, with recent changes, we always pick values for `affectedChannels` and `affectedChannelGroups` from `PubNub` instance - if it is ok, then I'm fine with it :)

Reply:
Ok, now I got your point I think! So you're proposing to have two more vars like:

```
var channels: [String]
var groups: [String]
```

And:

1. For the `.subscriptionChanged(channels, groups):
  - Assing  channels and groups from this event to local variables mentioned before
2. For other events:
   - Getting channels/groups from `pubnub.subscribedChannels`, etc

Is this what you meant? :)